app/components/avo/base_component.rb

```diff
@@ -57,12 +57,14 @@ def parent_or_child_resource
   end
 
   def link_to_child_resource_is_enabled?
-    return field_linked_to_child_resource? if @parent_resource
+    enabled = field_linked_to_child_resource? if @parent_resource
+    return enabled unless enabled.nil?
 
     @resource.link_to_child_resource
   end
 
   def field_linked_to_child_resource?
-    field.present? && field.respond_to?(:link_to_child_resource) && field.link_to_child_resource
+    resolved_field = @reflection ? @parent_resource.get_field(@reflection.name) : field
```

Comment:
I guess this will not find the field when using `for_attribute` option. For example:

```ruby
field :the_children, as: :has_many, for_attribute: :children, link_to_child_resource: true
```

This should cover that scenario, can you please verify that?

```suggestion
    resolved_field = @reflection ? @parent_resource.get_field(params[:for_attribute] || @reflection.name) : field
```